### PR TITLE
[Feature] Gitlab/Jenkins Pseudonyms

### PR DESCRIPTION
**Pull request by dfuchss**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] @ivanchimeno has added some test cases 

### Motivation and Context
GitLab and Jenkins allow any user to see any user's basic information. This leads to the problem that any student / user can map a pseudonym (login name) to the actual name of the user. This is a privacy issue. See also https://gitlab.com/gitlab-org/gitlab/-/issues/14123 

#### Jenkins
image

#### GitLab
image

### Description
I've added two new properties (this should be the most flexible way):
* gitlab.use-pseudonyms: false
* jenkins.use-pseudonyms: false

Enabling the resp. property forces artemis to send "User \<loginname\>" as user's name to gitlab or jenkins.

### Steps for Testing

1. Enable the properties (or only one iff you only want to test one system)
2. Login & create a new user
3. Check whether the new user's name is visible in jenkins resp. gitlab

### Test Coverage
I cannot see this.

If you have any questions, feel free to ask :)

### CODE REVIEW

File: src/main/resources/config/application-gitlab.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# ===================================================================
+# GitLab specific properties: this file will only be loaded during startup if the profile gitlab is active
+#
+# This configuration overrides the application.yml file.
+# ===================================================================
+
+gitlab:
+    # The following (optional) parameter allows to enable the use of pseudonyms.
+    # Iff activated, the user's first and last name are not transferred to gitlab.
```
Comment (krusche) [inline src/main/resources/config/application-gitlab.yml:9]:
```suggestion
    # If activated, the user's first and last name are not transferred to gitlab.
```

File: src/main/resources/config/application-jenkins.yml [hunk 2]
```diff
@@ -10,3 +10,8 @@ jenkins:
     # The default value for this parameter is true, so if it is not specified, crumb protection will be active.
     # If you experience problems using crumb, you can deactivate it here.
     use-crumb: true
+
+    # The following (optional) parameter allows to enable the use of pseudonyms.
+    # Iff activated, the user's first and last name are not transferred to jenkins.
```
Comment (krusche) [inline src/main/resources/config/application-jenkins.yml:15]:
```suggestion
    # If activated, the user's first and last name are not transferred to jenkins.
```

**Comment by dfuchss** [comment]
@ivanchimeno thanks for the tests :)

**Comment by ivanchimeno** [comment]
> @ivanchimeno thanks for the tests :)

@dfuchss I took the liberty and added additional tests improve the overall code coverage of the Jenkins user management service :)

**Review by ivanchimeno — APPROVED** [review]
Approving since the code looks good to me and the changes for both on my local environment and Ts2

**Review by krusche — APPROVED** [review]

**Pull request merged**